test/fakeintake/server/server.go

```diff
@@ -286,21 +308,31 @@ func (fi *Server) safeAppendPayload(route string, data []byte, encoding string)
 	if _, found := fi.payloadStore[route]; !found {
 		fi.payloadStore[route] = []api.Payload{}
 	}
-	fi.payloadStore[route] = append(fi.payloadStore[route], api.Payload{
+	rawPayload := api.Payload{
 		Timestamp: fi.clock.Now(),
 		Data:      data,
 		Encoding:  encoding,
-	})
+	}
+	fi.payloadStore[route] = append(fi.payloadStore[route], rawPayload)
+	fi.payloadParser.parse(rawPayload, route)
 }
 
-func (fi *Server) safeGetPayloads(route string) []api.Payload {
+func (fi *Server) safeGetRawPayloads(route string) []api.Payload {
 	fi.mu.Lock()
 	defer fi.mu.Unlock()
 	payloads := make([]api.Payload, 0, len(fi.payloadStore[route]))
 	payloads = append(payloads, fi.payloadStore[route]...)
 	return payloads
 }
 
+func (fi *Server) safeGetJsonPayloads(route string) []api.ParsedPayload {
+	fi.mu.Lock()
+	defer fi.mu.Unlock()
+	payloads := make([]api.ParsedPayload, 0, len(fi.payloadParser.payloadJsonStore[route]))
+	payloads = append(payloads, fi.payloadParser.payloadJsonStore[route]...)
```

Comment:
From an encapsulation point of view, it is not great to access field like that. Can you add a getter to `payloadParser.` that returns the JSON payload or an error if the route is not defined at https://github.com/DataDog/datadog-agent/pull/18332/files#diff-067626d001d8197c8320bdd344daa17d8459b1d48eeecc14bc74ce50877bedcdR67-R71